### DISCUSSION TITLE
GetMetrics grpc method is not called

### DISCUSSION THREAD

**Discussion by Jimyhjy** [kedacore/keda#4871] [Q&A / Need Help] [answered]
Why doesn't the GetMetrics method get called when using an external scaler, but all the other grpc methods work fine?  version 2.8.2

**Comment by tomkerkhove** [comment]
@zroubalik @JorTurFer Do you know?

@Jimyhjy can you provide some more details please?

**Reply by Jimyhjy** [reply]
I'm installing the yaml method of deployment from the documentation.

kubectl apply -f https://github.com/kedacore/keda/releases/download/v2.8.2/keda-2.8.2.yaml

The hpa is displayed as follows
![image](https://github.com/kedacore/keda/assets/14359380/d2e79316-6760-4e5b-b1c3-a975b46fbdfd)

But strangely enough, my deployment in another k8s cluster is fine

**Reply by Jimyhjy** [reply]
![image](https://github.com/kedacore/keda/assets/14359380/8583a66e-5316-4b83-83cb-11965a2de5d8)
this is scale object yaml file

**Reply by JorTurFer** [reply]
I don't get your point, where isn't called the method? In your gRPC server? In KEDA?
How are yoiu testing it? How many instances of the monitored workload do you have?
That method is called only during the HPA value requests, so it's not called during the workload is scaled to 0.
This call is done by KEDA metrics server in that version, do yuu see anything in the logs?

**Reply by Jimyhjy** [reply]
testing in my gRPC server,this is instances
![image](https://github.com/kedacore/keda/assets/14359380/5a84c376-2f32-4186-8aa7-69999203a8db)

metrics server logs
![image](https://github.com/kedacore/keda/assets/14359380/a7c07c12-53c7-4e00-ab6a-156fb7e87e6b)

**Reply by Jimyhjy** [reply]
deployment in another k8s cluster is fine, metrics server with the following extra content in the logs

![image](https://github.com/kedacore/keda/assets/14359380/bab3a7a9-dce6-461d-aa25-c3d38701de25)

Could it be a k8s cluster problem?

**Reply by Jimyhjy** [reply]
I know! this problem is in kube apiserver

![image](https://github.com/kedacore/keda/assets/14359380/e62ee427-8009-47e1-8de2-729a271f4536)


thank you!

**Reply by JorTurFer** [reply]
Maybe is there any blocker between kube api server and KEDA metrics server?
Is the issue solved?

**Reply by Jimyhjy** [reply, marked as answer]
yes，It's solved. thank you!
